pkg/cri/resource-manager/control/blockio/blockio.go

```diff
@@ -134,26 +134,51 @@ func (ctl *blockioctl) BlockIOClass(c cache.Container) string {
 		}
 	}
 
-	log.Debug("BlockIO class for %s (%s): %q", c.PrettyName(), cclass, blockioclass)
-
 	return blockioclass
 }
 
-// configNotify is blockio class mapping configuration notification callback.
+// configNotify is blockio class mapping and class definition configuration callback
 func (ctl *blockioctl) configNotify(event config.Event, source config.Source) error {
-	// BlockIO class mapping in opt (in flags.go) has changed.
-	// It will affect only new pods. cgroupsblkio parameters of running pods
-	// is not changed.
-	log.Info("class mapping configuration updated")
+	ignoreErrors := (event == config.RevertEvent)
+	err := blockio.UpdateOciConfig(ignoreErrors)
+	if err != nil && !ignoreErrors {
+		return err
+	}
+	// Possible errors in reconfiguring running containers are not errors in
+	// the updated configuration, therefore silently ignored.
+	ctl.reconfigureRunningContainers()
 	return nil
 }
 
+// reconfigureRunningContainers force setting current blockio configuration to all containers running on the node
+func (ctl *blockioctl) reconfigureRunningContainers() error {
+	var errors *multierror.Error
+	if ctl.cache == nil {
+		return nil
```

Comment:
I think that would make the control flow more readable.

So the order in the controller point of view would be:

1. `init()` by Go, no cache available
2. `Create(cache.Cache)`
3. All configuration notifications that are bound at `init` or possibly in `Create`. In this case `reconfigureRunningContainers()` could not called between `Create()` and `Start()`, because not all (both) necessary configurations are there.
4. `Start()` - here we know that all configurations are in place. In this case this would be a good spot to call `reconfigureRunningContainers()`.

So I think that I should still add `reconfigureRunningContainers()` to `Start()` anyway now, because currently `ctl.cache == nil` indicates that the controller has not been `Start()`ed, so `configNotify()` has not reconfigured anything yet.

Reply:
Yes. Having it in `Start()` is a good idea.